mthamayor/auto-mart
# Users and admins can filter cars

#### What does this PR do?
Users and admins can filter car adverts with one or more of the vehicle attributes

#### Description of Task to be completed?
Create the following API endpoints for users
1. GET /car?status=available 
2. GET /car?status=available&min_price=​XXXValue​&max_price=​XXXValue
3. GET /car?status=available&state=<new|used>
4. GET /car?status=available&manufacturer=​XXXValue
5. GET /car?body_type=​bodyType (Admin protected route)
6. GET /car (Admin protected route)

#### How should this be manually tested?
After cloning the repo

Start the development server with
`npm run dev`
On postman
If you want to access the protected routes, a token should be provided through the authorization header
`Bearer <token>`
the URLs are
`
http://localhost:3000/api/v1/car?status=available`
`http://localhost:3000/api/v1/car?status=available&min_price=​<XXXValue>​&max_price=​<XXXValue>`
`http://localhost:3000/api/v1/car?status=available&state=new`
`http://localhost:3000/api/v1/car?status=available&state=used`
`http://localhost:3000/api/v1/car?status=available&manufacturer=<​XXXValue>`
`http://localhost:3000/api/v1/car?body_type=​<bodyType>`
`http://localhost:3000/api/v1/car
`
Hit send on postman
You should receive a response similar to 
```
{
    "status": 200,
    "data": []
}
```
#### Any background context you want to provide?
No

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/166283864
https://www.pivotaltracker.com/story/show/166284213
https://www.pivotaltracker.com/story/show/166284431
https://www.pivotaltracker.com/story/show/166350002
https://www.pivotaltracker.com/story/show/166350113
https://www.pivotaltracker.com/story/show/166350182
